### PR TITLE
doc: Add ability to generate man pages

### PR DESCRIPTION
**Pull request by dcpleung**
For those who cannot really make the transition from command line to GUI (well... me...), add the ability to generate man pages so there can be a quicker way of finding out symbol meanings. Do this with:

* `BUILD_MANPAGES=1 make doxygen`

### CODE REVIEW

**Review by ceolin — DISMISSED** [review]

**Comment by nashif** [comment]
oooh man.... pages

File: doc/CMakeLists.txt [hunk 1]
```diff
@@ -18,6 +18,7 @@ set(LATEXMKOPTS "-halt-on-error -no-shell-escape" CACHE STRING "Default latexmk
 set(DT_TURBO_MODE OFF CACHE BOOL "Enable DT turbo mode")
 set(DOC_TAG "development" CACHE STRING "Documentation tag")
 set(DTS_ROOTS "${ZEPHYR_BASE}" CACHE STRING "DT bindings root folders")
+set(BUILD_MANPAGES OFF CACHE BOOL "Ask doxygen to also build man pages")
```
Comment (gmarull) [inline doc/CMakeLists.txt:21]: can you prefix this option with DOXYGEN since this is only for doxygen? Sphinx also supports man pages.

**Review by nashif — APPROVED** [review]

**Review by gmarull — APPROVED** [review]

**Review by henrikbrixandersen — APPROVED** [review]

**Comment by henrikbrixandersen** [comment]
@kartben This seems very isolated. Why not consider it for v3.7.0 LTS?

**Comment by kartben** [comment]
> @kartben This seems very isolated. Why not consider it for v3.7.0 LTS?

Yes, kinda isolated, but it is effectively a new feature in the doc generation build scripts :)

There is an easy and more than acceptable workaround which is that folks can manually tweak their Doxyfile configuration file, just like they would if there is anything else they'd want to tweak in the way Doxygen behave by default (ex. to change styling, etc.). I would also argue that it is probably not ideal to basically paraphrase Doxyfile config file like this PR is proposing as it doesn't scale: what happens when someone asks for Docbook or LaTeX?). We should maybe try to look at providing a more generic way to override *any* Doxyfile option.

**Review by kartben — CHANGES_REQUESTED** [review]
I am not sure about making man page generation a first class citizen, TBH.  Or, if we are, it should be exercised in CI, as we know from experience that there can easily be uncaught regressions somewhere in the doc generation pipeline.

Also, as per my latest comment in the PR thread, it feels as if adding new CMakeProperties that 1:1 map to Doxyfile entries won't really scale. It would be better IMO to provide a way for folks to easily override the default `zephyr.doxyfile.in` e.g by providing the path to a  `custom.doxyfile.in` that would extend the original one?

**Comment by dcpleung** [comment]
doxygen does not do overlay config files. This means that you will have to maintain separate config files and they are going to be out of sync. Or we can provide our own scripts to merge config files... which also has maintenance cost.

**Comment by kartben** [comment]
> doxygen does not do overlay config files. 

I was thinking something like adding an `@INCLUDE = custom.doxyfile.in` entry that would point to an empty file (path can even be customizable, I guess) that folks can then start adding custom entries to? I haven't tested it but hopefully something like this can do the trick?

Doxygen doc says:

> If the same tag is assigned more than once, the last assignment overwrites any earlier assignment.

So having the `@INCLUDE` at the bottom of the file should do?

**Comment by dcpleung** [comment]
Maybe something like https://github.com/zephyrproject-rtos/zephyr/pull/74511?

**Comment by kartben** [comment]
Superseded by #74511

**Pull request closed without merging**